### PR TITLE
Delivery form pickup

### PR DESCRIPTION
**Pull request by Pierre-LHOSTE**
https://github.com/coopcycle/coopcycle-app/issues/1792

| Pickup (default address) | Pickup (custom address) |
|-|-|
| Capture d’écran 2024-11-12 à 15 57 56 | Capture d’écran 2024-11-12 à 15 59 00 | 

| Dropoff | Time slot & packages |
|-|-|
| Capture d’écran 2024-11-12 à 15 58 21 | Capture d’écran 2024-11-12 à 15 59 21 |

### CODE REVIEW

File: src/navigation/store/NewDeliveryAddress.js [hunk 1]
```diff
@@ -167,6 +172,20 @@ function NewDeliveryAddress(props) {
         setFieldTouched,
       }) => (
         <ModalFormWrapper handleSubmit={handleSubmit} t={t}>
```
Comment (vladimir-8) [inline src/navigation/store/NewDeliveryAddress.js:174]: All strings should be localised

File: src/navigation/store/NewDeliveryPickup.js [hunk 2]
```diff
@@ -0,0 +1,425 @@
...
+          description: values.description,
+          businessName: values.businessName,
+          address,
+        }
+      : undefined;
+
+    navigation.navigate('StoreNewDeliveryAddress', { pickup });
+  }
+
+  return (
```
Comment (vladimir-8) [inline src/navigation/store/NewDeliveryPickup.js:160]: Same here, please more strings to i18n

File: src/navigation/store/NewDeliveryPickup.js [hunk 3]
```diff
@@ -0,0 +1,425 @@
...
+  errorText: {
+    color: '#FF4136',
+    marginTop: 5,
+  },
+  textarea: {
+    minHeight: 25 * 3,
+  },
+});
+
+function mapDispatchToProps(state) {
```
Comment (vladimir-8) [inline src/navigation/store/NewDeliveryPickup.js:406]: I'd suggest replacing these with `useSelector` and `useDispatch`

File: src/navigation/store/NewDeliveryPickup.js
Comment (vladimir-8) [inline src/navigation/store/NewDeliveryPickup.js:1]: a small misalignment:

Simulator Screenshot - iPhone 16 - 2024-11-08 at 10 32 06

Comment (vladimir-8) [inline src/navigation/store/NewDeliveryPickup.js:1]: saw this bug in the dark mode: 

Simulator Screenshot - iPhone 16 - 2024-11-08 at 11 12 10

Comment (vladimir-8) [inline src/navigation/store/NewDeliveryPickup.js:1]: Where did you get this object structure? It seems these fields are not saved currently. I guess all of them should be in the `Address` object. And `businessName` should go into `name` field

Screenshot 2024-11-08 at 10 42 29

File: src/navigation/store/NewDeliveryPickup.js [hunk 4]
```diff
@@ -0,0 +1,425 @@
...
+                  onBlur={handleBlur('address')}
+                  value={address}
+                  onSelectAddress={e => onSelectAddress(e, setFieldValue)}
+                  containerStyle={[
+                    {
+                      flex: 1,
+                      justifyContent: 'center',
+                    },
+                  ]}
+                  style={{
```
Comment (vladimir-8) [inline src/navigation/store/NewDeliveryPickup.js:245]: Should be without an extra array:
```
                  containerStyle={
                    {
                      flex: 1,
                      justifyContent: 'center',
                    }
                  }
```

**Review by vladimir-8 — CHANGES_REQUESTED** [review]

File: src/navigation/store/NewDeliveryPickup.js
Comment (Pierre-LHOSTE) [reply src/navigation/store/NewDeliveryPickup.js:1]: Issue that comes from the master

File: src/navigation/store/NewDeliveryPickup.js (hunk 3 above)
Comment (Atala) [reply src/navigation/store/NewDeliveryPickup.js:409]: const { t } = useTranslation()

**Review by vladimir-8 — APPROVED** [review]

**Comment by vladimir-8** [comment]
Looks good to me. Should we merge it?

**Pull request merged**